### DISCUSSION TITLE
Don't open [No Name] buffer when last buffer is closed

### DISCUSSION THREAD

**Discussion by ruckerzerg** [LazyVim/LazyVim#3515] [Q&A]
Whenever I close the last open buffer with `<leader>bd`, nvim will create a new [No Name] buffer that stays when I open a new file.
Can this behavior somehow be changed, so that when I close the last buffer the LazyVim startscreen is shown instead of a new [No Name] buffer?

Thanks for your help!

**Comment by dpetka2001** [comment]
That's default Neovim behavior. Don't know if you can get around that in some way I'm not aware of. But if you do `nvim --clean init.lua`, then you do `:bdelete` and then you do `:e init.lua` again, you will see the `[No Name]` buffer listed when you do `:buffers` as well.

**Comment by ruckerzerg** [comment]
Any idea how I could display the start screen when the last buffer is closed? I've seen people doing this with other Neovim distros.

**Reply by dpetka2001** [reply]
No idea, never done this before, because it's of no use to me. I only use Dashboard when I first open Neovim not after.

**Comment by mingming-cn** [comment]
This code can auto delete `No Name` buffer

```
vim.api.nvim_create_augroup("AutoDeleteNoNameBuffer", { clear = true })
vim.api.nvim_create_autocmd("BufEnter", {
  group = "AutoDeleteNoNameBuffer",
  pattern = "*",
  callback = function()
    local bufsize = #vim.fn.getbufinfo({ buflisted = 1 })
    if bufsize == 2 then
      for _, buf in pairs(vim.fn.getbufinfo({ buflisted = 1 })) do
        if buf.name == "" then
          vim.api.nvim_buf_delete(buf.bufnr, { force = true })
        end
      end
    end
  end,
})
```
